src/indexable.cr

```diff
@@ -5,6 +5,22 @@
 # `-2` is the next to last element, and so on.
 #
 # Types including this module are typically `Array`-like types.
+#
+# ## Stability guarantees
+#
+# Several methods in `Indexable`, such as `#bsearch` and `#cartesian_product`,
+# require the collection to be _stable_; that is, calling `#each(&)` over and
+# over again should always yield the same elements, provided the collection is
+# not mutated between the calls. In particular, `#each(&)` itself should not
+# mutate the collection throughout the loop. Stability of an `Indexable` is
+# guaranteed if the following criteria are met:
+#
+# * `#unsafe_fetch` and `#size` do not mutate the collection
+# * `#each(&)` and `#each_index(&)` are not overridden
```

Comment:
```suggestion
# * `#each(&)` and `#each_index(&)` are not overridden
```
Technically, this is not a strict requirement. It's only relevant that the behaviour of `each` and `each_index` is not altered. You *can* technically override them to use a different implementation, as long as it keeps these guarantees. Not sure if that's practically relevant for anything.

Reply:
This is why the paragraph says "if", not "if and only if".